src/bentoml/_internal/models/model.py

```diff
@@ -342,13 +344,22 @@ def to_runner(
         """
         from ..runner import Runner
 
+        # TODO: @larme @yetone run this branch only yatai version is incompatible with embedded runner
+        yatai_version = os.environ.get("YATAI_T_VERSION")
+        if embedded and yatai_version:
+            logger.warning(
+                f"Yatai of version {yatai_version} is incompatible with embedded runner, set `embedded=False` for runner {name}"
+            )
+            embedded = False
+
```

Comment:
This is for old version of yatai, which will ignore `BentoRunnerInfo`'s embedded field and spawn a pod anyway. In this case, we will still use non-embedded mode for embedded runner.

When yatai is compatible with embedded field, let's add a version check here. @yetone

Reply:
LGTM!